src/ShortcutEntry.vala

```diff
@@ -47,53 +47,10 @@ public class ShortcutEntry : Object {
         if (key_value.is_of_type (VariantType.ARRAY)) {
             string[] accels = key_value.get_strv ();
             if (accels.length > 0) {
-                parse_accelerator (accels[0]);
+                accels = Granite.accel_to_string (accels[0]).split (" + ");
```

Comment:
The accels you are writing to is conflicting with the Class's `accels` property. You should either do a `this.accels = ...` or change the variable name in line 48 :)